src/promptflow/promptflow/_sdk/entities/_connection.py

```diff
@@ -669,6 +669,26 @@ def __init__(
         super().__init__(configs=configs, secrets=secrets, **kwargs)
         self.module = kwargs.get("module", self.__class__.__module__)
         self.custom_type = custom_type or self.__class__.__name__
+        self.package = kwargs.get(CustomStrongTypeConnectionConfigs.PACKAGE, None)
+        self.package_version = kwargs.get(CustomStrongTypeConnectionConfigs.PACKAGE_VERSION, None)
+
+    def __getattribute__(self, item):
```

Comment:
Shall we override the __getattribute__, maybe __getattr__? getattribute is more dangerous than getattr, as getattr will be called when attribute is missing, it's a fall back op, but getattribute will be called every time when access an attribute.

Reply:
We have to call getattribute in this case.. Since we write the function this way:
```
class FakeClass:
field_1: str
field_2: str 
```

FakeClass.field_1 will not enter **getattr**

Reply:
added comment over the function